cli/src/commands/content/updateChannel.ts

```diff
@@ -112,21 +96,44 @@ export default class UpdateChannelCommand extends UploadCommandBase {
     meta.coverPhoto = assetIndices.coverPhotoPath
     meta.avatarPhoto = assetIndices.avatarPhotoPath
 
-    // Preare and send the extrinsic
+    // Prepare and send the extrinsic
+    const serializedMeta = metadataToBytes(ChannelMetadata, meta)
+    const expectedDataObjectStateBloatBond = await this.getApi().dataObjectStateBloatBond()
     const assetsToUpload = await this.prepareAssetsForExtrinsic(resolvedAssets)
     const assetsToRemove = await this.getAssetsToRemove(
       channelId,
       assetIndices.coverPhotoPath,
       assetIndices.avatarPhotoPath
     )
 
-    const expectedDataObjectStateBloatBond = await this.getApi().dataObjectStateBloatBond()
+    // Ensure actor is authorized to perform channel update
+    const requiredPermissions: ChannelActionPermission['type'][] = []
+    if (collaborators) {
+      requiredPermissions.push('ManageChannelCollaborators')
+    }
+    if (assetsToUpload) {
+      requiredPermissions.push('ManageNonVideoChannelAssets')
+    }
+    if (serializedMeta.length) {
+      requiredPermissions.push('UpdateChannelMetadata')
+    }
+    if (!(await this.hasRequiredChannelAgentPermissions(actor, channel, requiredPermissions))) {
+      this.error(
+        `Only channelOwner or collaborator with ${requiredPermissions} permission can update channel ${channelId}!`,
+        {
+          exit: ExitCodes.AccessDenied,
+        }
+      )
+    }
+
     const channelUpdateParameters = createType('PalletContentChannelUpdateParametersRecord', {
-      assetsToUpload: assetsToUpload,
       expectedDataObjectStateBloatBond,
-      collaborators: [],
-      assetsToRemove: createType('BTreeSet<u64>', assetsToRemove),
-      newMeta: metadataToBytes(ChannelMetadata, meta),
+      assetsToUpload,
+      assetsToRemove,
+      newMeta: serializedMeta,
+      collaborators: new Map(
+        collaborators?.map(({ memberId, channelAgentPermissions }) => [memberId, channelAgentPermissions])
+      ),
```

Comment:
Same issue as with `newMeta`:
```suggestion
      collaborators: collaborators?.length
        ? new Map(collaborators?.map(({ memberId, channelAgentPermissions }) => [memberId, channelAgentPermissions]))
        : null,
```